hatice-ecevit-2016400138/assignment3/Dict.java

```diff
@@ -0,0 +1,84 @@
+import java.util.HashMap;
+
+public class Dict {
+    public HashMap<Character,DictNode> letters = new HashMap<Character, DictNode>();
+
+    public void insertWord (String s){
+        if (!letters.containsKey(s.charAt(0))){
+            letters.put(s.charAt(0),new DictNode());
+        }
+        DictNode current = letters.get(s.charAt(0));
+        s = s.substring(1);
+        insertWord(s,current);
+    }
+
+    /**
+     *
+     * @param s is the word that is inserted.
+     * @param node is the current node in the dictionary tree.
+     */
+    private void insertWord (String s, DictNode node) {
+        if (node.getChildren().containsKey(s.charAt(0))) {
+            node = node.getChildren().get(s.charAt(0));
+        }
+        else {

```

Comment:
You have repeated code in your if and else branch. Can you restructure to remove the repeated code? Code repetition (copy and paste) is often the cause of maintenance bugs (as one side gets fixed and the other doesn't) so it is good practice to try and remove it.